contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -1,26 +1,148 @@
-        _mint(_asset, _amount, _minimumOusdAmount);
+        require(_amount > 0, "Amount must be greater than 0");
+        require(
+            _amount >= _minimumOusdAmount,
+            "Mint amount lower than minimum"
+        );
+
+        emit Mint(msg.sender, _amount);
+
+        // Rebase must happen before any transfers occur.
+        if (!rebasePaused && _amount >= rebaseThreshold) {
+            _rebase();
+        }
+
+        // Mint oTokens
+        oUSD.mint(msg.sender, _amount);
+
+        // Transfer the deposited coins to the vault
+        IERC20(_asset).safeTransferFrom(msg.sender, address(this), _amount);
+
+        // Auto-allocate if necessary
+        if (_amount >= autoAllocateThreshold) {
+            _allocate();
+        }
+    }
+
+    // @inheritdoc VaultCore
+    function _calculateRedeemOutputs(uint256 _amount)
+        internal
+        view
+        virtual
+        override
+        returns (uint256[] memory outputs)
+    {
+        // Overrides `VaultCore._calculateRedeemOutputs` to redeem with only
+        // WETH instead of LST-mix. Doesn't change the function signature
+        // for backward compatibility
+
+        // Calculate redeem fee
+        if (redeemFeeBps > 0) {
+            uint256 redeemFee = _amount.mulTruncateScale(redeemFeeBps, 1e4);
+            _amount = _amount - redeemFee;
+        }
+
+        // Ensure that the WETH index is cached
+        uint256 _wethAssetIndex = wethAssetIndex;
+        require(
+            allAssets[_wethAssetIndex] == weth,
+            "WETH Asset index not cached"
+        );
+
+        outputs = new uint256[](allAssets.length);
+        outputs[_wethAssetIndex] = _amount;
+    }
+
+    // @inheritdoc VaultCore
+    function _redeem(uint256 _amount, uint256 _minimumUnitAmount)
+        internal
+        virtual
+        override
+    {
+        // Override `VaultCore._redeem` to simplify it. Gets rid of oracle
+        // usage and looping through all assets for LST-mix redeem. Instead
+        // does a simple WETH-only redeem.
+        emit Redeem(msg.sender, _amount);
+
+        if (_amount == 0) {
+            return;
+        }
+
+        // Amount excluding fees
+        uint256 amountMinusFee = _calculateRedeemOutputs(_amount)[
+            wethAssetIndex
+        ];
+
+        require(
+            amountMinusFee >= _minimumUnitAmount,
+            "Redeem amount lower than minimum"
+        );
+
+        if (IERC20(weth).balanceOf(address(this)) >= amountMinusFee) {
+            // Use Vault funds first if sufficient
+            IERC20(weth).safeTransfer(msg.sender, amountMinusFee);
+        } else {
+            address strategyAddr = assetDefaultStrategies[weth];
+            if (strategyAddr != address(0)) {
+                // Nothing in Vault, but something in Strategy, send from there
+                IStrategy strategy = IStrategy(strategyAddr);
+                strategy.withdraw(msg.sender, weth, amountMinusFee);
```

Comment:
It is probably necessary to first withdraw from the vault and then withdraw what is missing from the strategy. Otherwise `revert("Liquidity error");` might trigger even if there is enough weth on vault and strategy combined.

Reply:
Agree that we might want to move to a more sophisticated system here eventually (if we are spending the gas to pull funds out, we might want to exactly rebalance our vault buffer for example.) For the time being I think we'll keep this PR small and focused and continue with the code previous code that the vault has been using.

Reply:
Codewise it's not a lot more sophisticated to pull vault balance if there is any and then take what is missing from the strategy. You are saving gas by not having to `_allocate` the pending weth if any.